proxy/proxy.go

```diff
@@ -51,19 +61,41 @@ func main() {
 
 	loggedRouter := handlers.LoggingHandler(os.Stdout, proxyRouter)
 
-	err = http.ListenAndServe(":"+c.Port, loggedRouter)
+	err = http.ListenAndServe(":" + c.Port, loggedRouter)
 	if err != nil {
 		fmt.Println(err)
 	}
 }
-
+func CallbackUrl() (string, error) {
+	var vcapApp VcapApplication
+	err := json.Unmarshal([]byte(c.VcapAppRaw), &vcapApp)
+	if err != nil {
+		return "", err
+	}
+	// we try first ssl to see if it can be used
```

Comment:
Sounds good, would you mind making that change? Also, I'm still not sure why this works--am I misreading, or is the proxy app trying to make a request to itself before it starts?